### PR TITLE
Add new plugin setting to toggle button visibility in team sidebar

### PR DESCRIPTION
**Pull request by colorfusion**
#### Summary

This pull request will implement a new system console setting to toggle the visibility of the buttons in the team sidebar, in the toolbar on the left hand side.  
The request will register a new WebSocket event to listen to config update events.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-plugin-todo/issues/109

### CODE REVIEW

File: server/plugin.go [hunk 1]
```diff
@@ -395,6 +400,36 @@ func (p *Plugin) handleBump(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	requestMethod := r.Method
+
+	if requestMethod != http.MethodGet {
+		http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
+		return
+	}
+
+	if p.configuration != nil {
```
Comment (larkox) [inline server/plugin.go:418]: The configuration may have in the future "secret" information that should not reach all the users. Could you pass here only the sidebar buttons related config?
I am all in for making this extensible in the future (create a struct like "client config" which now will only have the setting to hide the buttons, and a function "configToClientConfig" that just copy the relevant values), but at the moment it is not needed. Feel free to do it the way you prefer.

File: server/plugin.go [hunk 2]
```diff
@@ -403,6 +438,15 @@ func (p *Plugin) sendRefreshEvent(userID string) {
 	)
 }
 
+// Publish a WebSocket event to update the configuration of the application on the webapp end.
+func (p *Plugin) sendConfigUpdateEvent() {
+	p.API.PublishWebSocketEvent(
+		WSEventConfigUpdate,
+		nil,
```
Comment (larkox) [inline server/plugin.go:445]: Why not send the new configuration as payload?

**Comment by colorfusion** [comment]
@larkox Thanks for the feedback! I've made changes to the code based on the feedback, do let me know if you have any additional feedback.

File: webapp/src/selectors.js [hunk 3]
```diff
@@ -48,3 +48,4 @@ export const getSiteURL = () => {
 };
 
 export const isRhsVisible = (state) => getPluginState(state).isRhsVisible;
+export const isTeamSidebarHidden = (state) => getPluginState(state).isTeamSidebarHidden;
```
Comment (mickmister) [inline webapp/src/selectors.js:51]: Can we have this selector be `isTeamSidebarVisible`? It's fine to have the config and reducer be the `hidden` version, but I think for the selector it makes sense to use `visible`.

File: webapp/src/action_types.js [hunk 4]
```diff
@@ -8,3 +8,4 @@ export const GET_IN_ISSUES = pluginId + '_get_in_issues';
 export const RECEIVED_SHOW_RHS_ACTION = pluginId + '_show_rhs';
 export const UPDATE_RHS_STATE = pluginId + '_update_rhs_state';
 export const SET_RHS_VISIBLE = pluginId + '_set_rhs_visible';
+export const SET_HIDE_TEAM_SIDEBAR_BUTTON = pluginId + '_set_hide_team_sidebar';
```
Comment (mickmister) [inline webapp/src/action_types.js:11]: I suggest `SET_HIDE_TEAM_SIDEBAR_BUTTONS` as there are multiple buttons we're hiding

**Comment by colorfusion** [comment]
Thanks for the feedback @mickmister! I've committed the changes based on the suggestions made, do let me know if there are any other feedback / suggestions that you may have.

File: server/configuration.go [hunk 5]
```diff
@@ -81,7 +87,13 @@ func (p *Plugin) OnConfigurationChange() error {
 		return errors.Wrap(err, "failed to load plugin configuration")
 	}
 
+	sendEvent := p.hasClientConfigChanged(p.configuration, configuration)
```
Comment (mickmister) [inline server/configuration.go:90]:
```suggestion
	shouldUpdateClient := p.hasClientConfigChanged(p.configuration, configuration)
```

**Comment by colorfusion** [comment]
@mickmister I've committed a small fix to check for nil variables as `p.configuration` does not exist on the first call to `OnConfigurationChange`. My apologies for missing that out during the implementation.

**Pull request merged**